torch/csrc/jit/tracer.cpp

```diff
@@ -177,6 +185,19 @@ void ArgumentStash::stashIntListElem(const std::string& arg_name, size_t size, s
   list_trace[idx] = prim;
 }
 
+void ArgumentStash::stashValue(const std::string& arg_name, size_t idx, const Variable& var, TypePtr type) {
+  if (!isTracing()) return;
+
+  Value* ten = getValueTrace(var);
+  if (type) {
+    auto& g = *ten->owningGraph();
+    ten = g.createTensorToNum(type, ten)
```

Comment:
Well this code assumes that `type` is a numeric type, so you better at least put a comment somewhere. Apart from this, I really do believe that we should do the casting in `addInputs` and not in here. You're claiming that this method is for "Values that don't need special handling" after all.